code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+# Approximate run time: TBC
+##########################################################
+ 
+
+### 1.Housekeeping ----
+# This section should be the only section of the script which requires manual changes 
+# for future updates and includes:
+#   loading package_grps
+#   setting filepaths and extract dates
+#   functions (defined here or sourced from another file)
+#   setting plot parameter
+#   specifying codes (e.g. ICD-10 codes)
+
+
+
+### Filepaths for latest monthflag # may NOT NEEDED as now within set-up_environment ( check with Russell)
+
+filepath<-("/conf/delayed_discharges/RAP development/2020_02/Outputs/")
+filepath2<-("/conf/delayed_discharges/RAP development/2020_02/Data/scotland/")
+
+### 2. Bring in environment ----
+#( ran separately in code folder - how do i bring it in without copying / pasting as ?)
+source("00_setup_environment.R")
+
+#add in new library needed that isn't in setup.
+library(xlsx)
+
+# monthflagflag<-("Apr 2020")
+
+### 3. Get Scotland_validated file for latest monthflag ----
+
+datafile<-read.csv(paste0(filepath2,"SCOTLAND_validated.csv"))
+#sum(datafile$obds_in_month) check this matches publication when testing prevous data.
+
+#strip out code 100s
+
+
+datafile<-datafile %>% filter(datafile$reas1!="Code 100")
+#table(datafile$reas1) # check code 100 has been removed
+
+#Create new variables
+
+datafile<-datafile %>% mutate(areaname=" ")
+datafile<-datafile %>% mutate(year="2019-20")
+
+### 4a. Rename variables ----
+#( may not be needed once full RAP programs used as variables will have been renamed in earlier process)
+# datafile <- datafile %>%
+# rename(nhs_board=Healthboard,
+#          chi_number=CHINo,
+#          age_at_rdd=AGEATRDD,
+#          postcode=PatientPostcode,
+#          local_authority_code=LocalAuthorityArea,
+#          date_of_birth=PatientDOB,
+#          discharge_specialty_nat_code=SpecialtyCode,
+#          date_referred_for_sw_assessment=DateReferralReceived,
+#          date_declared_medically_fit=Readyfordischargedate,
+#          REASONFORDELAY=REASONFORDELAY,
+#          dd_code_2=REASONFORDELAYSECONDARY,
+#          out_of_area_case_indicator=Outofareacaseindicator,
+#          admission_date=OriginalAdmissionDate,
+#          sex_code=Gender,
+#          discharge_date=DateDischarge,
+#          discharge_to_code=DischargeReason,
+#          census_flag=CENSUSFLAG,
+#          delay_description=DELAY_DESCRIPTION)
+
+#Create age_grpgGrouping
+
+datafile<-datafile%>% mutate(age_grp=
+                               if_else(AGEATRDD<75, "18-74",
+                                       if_else(AGEATRDD>=75, "75+", " ")))
+
+
+
+
+
+#save out file for main bed days file
+ 
+
+#write_sav(datafile,paste0(filepath,"main bed days file.sav")) # save out file
+
+#set function
+#'%!in%' <- function(x,y)!('%in%'(x,y))
+
+#Save census data file
+
+
+datafile2 <- filter(datafile, !REASONFORDELAYSECONDARY %in%c("26X","46X") & CENSUSFLAG=="Y")
```

Comment:
See "Incremental Data-Frame Names" section in general feedback comments.
```suggestion
datafile <- filter(datafile, !(reasonfordelaysecondary %in% c("26X", "46X")) & census_flag == 1)
```